ui/src/common/actions.ts

```diff
@@ -473,31 +473,35 @@ export const StateActions = {
       // it's a group summary track that has a fixed explicit ID.
       // Note that (some, at least) summary tracks don't reference
       // their group
-      const id = track.trackGroup !== SCROLLING_TRACK_GROUP
-              && Object.values(state.trackGroups).some(group => group.tracks.length && group.tracks[0] === track.id)
-          ? { id: track.id }
-          : {};
-      
+      const id = track.trackGroup !== SCROLLING_TRACK_GROUP &&
+              Object.values(state.trackGroups).some((group) => group.tracks.length && group.tracks[0] === track.id) ?
+          {id: track.id} :
+          {};
+
+      // sortKey from the passed state was not resolved so I used global's
+      // The error was saying something about getting Proxy
+      // Which I noticed sortKey was set to
+      const sortKey = globals.state.tracks[args.trackId].trackSortKey;
```

Comment:
This looks really fishy, reaching into the previously committed global state from this processing of a draft.

In commit 77d1800f6622ff384e6411427974f078fc05840c on the multi-perfetto branch I propose moving the `filteredTracks` array from `Globals` into the `State`, which I think better fits the Perfetto approach to managing track-related information and may help with this problem by keeping the state (no pun intended) more consistent.

If you try cherry-picking that change instead of doing this `globals` access here, does it help?